### PR TITLE
Issue #345 part2 imod refactor primod update

### PR DESCRIPTION
**Pull request by JoerivanEngelen**
Fixes https://github.com/Deltares/imod_coupler/issues/345

This PR introduces the following things to primod:

- Provide MODFLOW6 objects in primod at write, instead of at init.
- Introduce a ``ModflowMixin``, which can be used to add methods for MODFLOW-related couplings. At the moment all couplings use MODFLOW, but there might be a Ribasim-WFLOW coupling in the future, for which MODFLOW methods are unnecessary. It now contains one static method.
- Add unittests for ModflowMixin
- Update fixtures and tests
- Update a fixture which calls an internal key "max_abstraction_surfacewater_m3_d", which is changed to "max_abstraction_surfacewater" in iMOD Python

- Boyscouting: Change calls from _get_pkg_key("dis") to _get_dis_key(), to also support getting imod.mf6.VerticesDiscretization in the future, for unstructured grids.

### CODE REVIEW

**Comment by JoerivanEngelen** [comment]
I see ribasim python crashing with Python 3.10 on a Pandera issue, I think this is irrelevant to this changeset. This also happened with the previous changeset. I'll fix this in the final merge.

File: pre-processing/primod/model_mixin.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+from imod.mf6 import Modflow6Simulation, StructuredDiscretization
+from imod.mf6.mf6_wel_adapter import Mf6Wel
+
+
+class ModflowMixin:
+    """Modflow specific methods to add to coupling objects."""
+
+    @staticmethod
+    def get_mf6_pkgs_with_coupling_dict(
```
Comment (HendrikKok) [inline pre-processing/primod/model_mixin.py:11]: Could you make this name more specific? We also couple to RIV, DRN and API packages in the Ribasim coupling for example. Those are not included here. Perhaps something like: get_mf6_pkgs_for_metaswap_coupling

File: tests/test_imod_coupler/test_ribametamod_cases.py [hunk 2]
```diff
@@ -88,12 +87,8 @@ def set_confined_storage_formulation(
     return mf6_model
 
 
-def remove_sprinkling_from_groundwater(
-    msw_model: MetaSwapModel, mf6_model: Modflow6Simulation
-) -> MetaSwapModel:
+def remove_sprinkling_from_groundwater(msw_model: MetaSwapModel) -> MetaSwapModel:
     msw_model.pop("sprinkling")
-    msw_model.pop("mod2svat")
-    msw_model["mod2svat"] = msw.CouplerMapping(modflow_dis=mf6_model["GWF_1"]["dis"])
```
Comment (HendrikKok) [inline tests/test_imod_coupler/test_ribametamod_cases.py:96]: Why did you remove this? Since the CouplerMapping was initialized with sprinkling activated, it includes those couplings. Thats why it is reinitialized.

**Review by HendrikKok — CHANGES_REQUESTED** [review]
Nice work, I have one small request on a name, and one question. Also, the test bench is not green right?

File: tests/test_imod_coupler/test_ribametamod_cases.py (hunk 2 above)
Comment (JoerivanEngelen) [reply tests/test_imod_coupler/test_ribametamod_cases.py:96]: Because the CouplerMapping doesn't take any arguments anymore. The modflow packages are now requested upon writing, instead of upon initialization, in iMOD Python. In pseudo-code:

Before the refactor in iMOD Python:

```python
class CouplerMapping:
    def __init__(self, modflow_dis, modflow_wel=None):
         ...
    def write(self):
         ...    
```

After it:

```python
class CouplerMapping:
    def __init__(self):
         ...
    def write(self,modflow_dis, modflow_wel=None):
         ...    
```

Therefore the CouplerMapping object doesn't need to be re-initialized: mappings are created upon writing.

File: pre-processing/primod/model_mixin.py (hunk 1 above)
Comment (JoerivanEngelen) [reply pre-processing/primod/model_mixin.py:11]: Good point! I didn't consider the Ribasim coupling when naming the method. Will fix.

**Comment by JoerivanEngelen** [comment]
> Nice work, I have one small request on a name, and one question. Also, the test bench is not green right?

See my previous comment ;-)

> I see ribasim python crashing with Python 3.10 on a Pandera issue, I think this is irrelevant to this changeset. This also happened with the previous changeset. I'll fix this in the final merge of the feature branch to main.

File: pre-processing/primod/model_mixin.py (hunk 1 above)
Comment (JoerivanEngelen) [reply pre-processing/primod/model_mixin.py]: I followed your suggestion for the method name, and renamed ModflowMixin to MetaModMixin, as the method is not used for RibaMod.

**Review by HendrikKok — APPROVED** [review]
